src/Module/Starred.php

```diff
@@ -10,51 +10,36 @@
  */
 class Starred extends BaseModule
 {
-	public static function rawContent()
+	public static function rawContent(array $parameters = [])
 	{
-		$a = self::getApp();
-		$starred = 0;
-		$itemId = null;
-
 		if (!local_user()) {
 			exit();
 		}
 
-		// @TODO: Replace with parameter from router
-		if ($a->argc > 1) {
-			$itemId = intval($a->argv[1]);
-		}
-
-		if (!$itemId) {
-			exit();
+		if (empty($parameters['item'])) {
+			exit;
+		} else {
+			$itemId = intval($parameters['item']);
 		}
 
 		$item = Item::selectFirstForUser(local_user(), ['starred'], ['uid' => local_user(), 'id' => $itemId]);
 		if (empty($item)) {
 			exit();
 		}
 
-		if (!intval($item['starred'])) {
-			$starred = 1;
-		}
+		$starred = !(bool)$item['starred'];
 
 		Item::update(['starred' => $starred], ['id' => $itemId]);
 
 		// See if we've been passed a return path to redirect to
 		$returnPath = $_REQUEST['return'] ?? '';
-		if ($returnPath) {
-			$rand = '_=' . time();
-			if (strpos($returnPath, '?')) {
-				$rand = "&$rand";
-			} else {
-				$rand = "?$rand";
-			}
-
-			$a->internalRedirect($returnPath . $rand);
+		if (!empty($returnPath)) {
+			$rand = '_=' . time() . (strpos($returnPath, '?') ? '&' : '?') . 'rand';
+			self::getApp()->internalRedirect($returnPath . $rand);
 		}
 
 		// the json doesn't really matter, it will either be 0 or 1
-		echo json_encode($starred);
+		echo json_encode((int)$starred);
 		exit();
```

Comment:
Please use `System::jsonExit` instead.

Reply:
For some unknown reason the jsonExit hadn't worked. I'm unsure why, so I changed that back.

Reply:
But `System::jsonExit` is just `echo json_encode($data); exit;`!?